### PR TITLE
Load upcoming events on example page

### LINKED ISSUE #648: Create an example page to display upcoming onelive events
**Is your feature request related to a problem? Please describe.**
Need to test the upcoming onelive events section

**Describe the solution you'd like**
Create an example page and display upcoming onelives.
*branch - "onelive-events"*

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by Piumal1999**
## Purpose

The purpose of this PR is to fix #648

## Goals

Load upcoming events on example page for testing purposes.

## Approach

Updated the component to hold a link
Got data in JSON format using Calendar API of sef-data-holder 
Used Mustache.js to render
Loaded upcoming events on a duplicate onelive page

### Screenshots

8b4d5339-22a4-4c52-b2c7-202e2a98f80d
  
### Preview Link

https://pr-649-sef-site.surge.sh/onelive/

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

## Related PRs
 
N/A

## Test environment
 
Ubuntu 19.04
Google Chrome
JRE: 1.8.0_202-release-1483-b44 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o

## Learning

N/A

### CODE REVIEW

File: src/main/webapp/onelive/index.html [hunk 1]
```diff
@@ -277,13 +285,75 @@ <h5 class="title">
...
+        loadUpcomingEvents(); //load upcoming events
         loadProfiles(); // load profiles
         loadFaqComponent();
     });
 
+    //function to load upcoming events
+    function loadUpcomingEvents() {
+        $.ajax({
+            type: 'GET',
+            url: 'http://localhost:8080/calendar/onelive',
```
Comment (Gravewalker666) [inline src/main/webapp/onelive/index.html]: localhost:8080?
Isn't this service up and running?

File: src/main/webapp/onelive/index.html [hunk 2]
```diff
@@ -277,13 +285,75 @@ <h5 class="title">
...
+                        payload.items[i].facebookLink = description.substring(description.lastIndexOf('<a href') + 9, description.lastIndexOf('" id='));
+                        payload.items[i].description = description.substring(0, description.lastIndexOf('<a href'));
+                        //get date & time of the event
+                        let start = new Date(payload.items[i].start.dateTime);
+                        let end = new Date(payload.items[i].end.dateTime);
+                        payload.items[i].day = start.toString().substring(0, 3);
+                        payload.items[i].month = start.toString().substring(4, 7);
+                        payload.items[i].date = start.toString().substring(8, 10);
+                        payload.items[i].startTime = start.toString().substring(16, 21);
+                        payload.items[i].endTime = end.toString().substring(16, 21);
```
Comment (Gravewalker666) [inline src/main/webapp/onelive/index.html]: Try .toLocaleDateString() and .toLocaleTimeString()

Comment (Gravewalker666) [reply src/main/webapp/onelive/index.html]: Not necessary but check those methods out.

**Review by Gravewalker666 — CHANGES_REQUESTED** [review]

File: src/main/webapp/onelive/index.html (hunk 1 above)
Comment (Piumal1999) [reply src/main/webapp/onelive/index.html]: Oh yes. I forgot to change it. I'll update

File: src/main/webapp/onelive/index.html (hunk 2 above)
Comment (Piumal1999) [reply src/main/webapp/onelive/index.html]: Ok I'll check

Comment (Piumal1999) [reply src/main/webapp/onelive/index.html]: It works! Thanks :star_struck:

**Review by Gravewalker666 — APPROVED** [review]

File: src/main/webapp/onelive/index.html [hunk 3]
```diff
@@ -277,13 +285,75 @@ <h5 class="title">
...
+                        payload.items[i].startTime = start.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit', hour12: false});
+                        payload.items[i].endTime = end.toLocaleTimeString([], {hour: '2-digit', minute:'2-digit', hour12: false});
+                        //Mustache render
+                        let content = Mustache.render($("#upcoming-events-template").html(), {"data": payload.items});
+                        $("#upcoming-events").html(content);
+                    }
+                }
+            },
+            error: function () {
+                console.log("Error occurred while loading upcoming events");
```
Comment (jayasanka-sack) [inline src/main/webapp/onelive/index.html:351]: Console logs should not be in the code.

**Comment by jayasanka-sack** [comment]
The preview isn't working
Screenshot 2020-04-11 at 15 18 03

File: src/main/webapp/examples/components.html [hunk 4]
```diff
@@ -2104,26 +2104,30 @@ <h2 class="mb-5">
...
-                            <span class="mr-5"> 7.00 - 8.00 pm</span>
-                            <span class="font-weight-bold ml-1">Energy harvesting in wearable electronics</span>
+                    <a href="" target="_blank">
+                        <div class="card card-event shadow shadow-lg--hover mb-1">
+                            <div class="card-body">
+                                <span class="h5 mr-1 card-event-date">12</span>
+                                <span class="small text-muted mr-4 text-capitalize">APR, SUN</span>
+                                <span class="mr-5"> 7.00 - 8.00 pm</span>
+                                <span class="font-weight-bold ml-1">Energy harvesting in wearable electronics</span>
+                            </div>
```
Comment (jayasanka-sack) [inline src/main/webapp/examples/components.html:2114]: Is this a dummy event?

Comment (Piumal1999) [reply src/main/webapp/examples/components.html:2114]: Yes

File: src/main/webapp/onelive/index.html (hunk 3 above)
Comment (Piumal1999) [reply src/main/webapp/onelive/index.html:351]: The event in the calendar is not in the suitable format. It was added as full day event. Have to add the events as limited time events

> The preview isn't working
> Screenshot 2020-04-11 at 15 18 03

The event in the calendar is not in the suitable format. It was added as full day event. Have to add the events as limited time events

**Comment by Piumal1999** [comment]
> The preview isn't working
> Screenshot 2020-04-11 at 15 18 03

The event in the calendar is not in the suitable format. It was added as full day event. Have to add the events as limited time events

File: src/main/webapp/onelive/index.html (hunk 3 above)
Comment (Piumal1999) [reply src/main/webapp/onelive/index.html:351]: Ok

**Pull request merged**